services/spar/src/Spar/API.hs

```diff
@@ -112,6 +117,16 @@ api opts =
 appName :: ST
 appName = "spar"
 
+
+authreq :: NominalDiffTime -> Maybe URI.URI -> Maybe URI.URI -> SAML.IdPId -> Spar (SAML.FormRedirect SAML.AuthnRequest)
+authreq authreqttl msucc merr idpid = do
+  form@(SAML.FormRedirect _ ((^. SAML.rqID) -> reqid)) <- SAML.authreq authreqttl idpid
+  wrapMonadClient . Data.storeVerdictFormat authreqttl reqid =<< case (msucc, merr) of
+    (Nothing, Nothing) -> pure VerdictFormatWeb
+    (Just ok, Just err) -> pure $ VerdictFormatMobile ok err
+    _ -> throwSpar SparBadInitiateLoginQueryParams
```

Comment:
-  validation before request construction is an easy change, done.

- substitution of the URI is not possible here because it requires cookie and userid, which are only constructed during finalize-login.  the errors that this can trigger later are internal errors in the client code, since the urls are (more or less) hard-coded there, and not entered or affected by the user.  not changing anything.

- Length and schema checks are not in the scope of this PR (we're currently blocking mobile teams with delaying this), and it is on the TODO list, so we won't forget it.  i'll try to make some progress on this front now, though, as it's simple enough.  next PR?